src/azure-cli/azure/cli/command_modules/acr/_params.py

```diff
@@ -430,8 +430,9 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
         c.argument('description', options_list=['--description'], help='Description for the scope map. Maximum 256 characters are allowed.', required=False)
         c.argument('scope_map_name', options_list=['--name', '-n'], help='The name of the scope map.', required=True)
 
-    repo_valid_actions = "Valid actions are {}".format({action.value for action in RepoScopeMapActions})
-    gateway_valid_actions = "Valid actions are {}".format({action.value for action in GatewayScopeMapActions})
+    # Action strings generated this way to ensure consistent ordering each time the help text is generated.
+    repo_valid_actions = "Valid actions are {" + ', '.join('\'{}\''.format(action.value) for action in RepoScopeMapActions) + "}."
+    gateway_valid_actions = "Valid actions are {" + ', '.join('\'{}\''.format(action.value) for action in GatewayScopeMapActions) + "}."
```

Comment:
Just curiosity, may I ask why this way can solve the problem of random order?

Reply:
I really don't understand why this way gives a defined order and the other way doesn't. I would have thought that the original code would have a defined order too, but it doesn't